modules/afsql/afsql.c

```diff
@@ -805,15 +807,93 @@ afsql_dd_build_insert_command(AFSqlDestDriver *self, LogMessage *msg, GString *t
             }
           else
             {
-              dbi_conn_quote_string_copy(self->dbi_ctx, value->str, &quoted);
-              if (quoted)
+              TypeHint type = self->fields[i].value->type_hint;
+              switch(type)
+                {
+                case TYPE_HINT_INT32:
+                case TYPE_HINT_INT64:
+                {
+                  gint64 k;
+                  if (type_cast_to_int64(value->str, &k, NULL))
+                    {
+                      dbi_conn_escape_string_copy(self->dbi_ctx, value->str, &quoted);
```

Comment:
`quoted` should be freed after each successful `dbi_conn_escape_string_copy()` call.
I'd also consider using another local variable, `escaped`, for example, so it's clear that type-hinted values are only escaped, they are not quoted.

(Currently, only `dbi_conn_quote_string_copy()` are freed at the bottom of this function.)